Library/Homebrew/cask/lib/hbc/quarantine.rb

```diff
@@ -0,0 +1,57 @@
+require "development_tools"
+module Hbc
+  module Quarantine
+    module_function
+
+    QUARANTINE_ATTRIBUTE = "com.apple.quarantine".freeze
+
+    # @private
+    def swift
+      @swift ||= DevelopmentTools.locate("swift")
+    end
+
+    def available?
+      status = !swift.nil?
+      odebug "Quarantine is #{status ? "available" : "not available"}."
+      !swift.nil?
+    end
+
+    def detect(file)
+      odebug "Verifying Gatekeeper status of #{file}"
+
+      quarantine_status = !Quarantine.status(file).empty?
+
+      odebug "#{file} is #{quarantine_status ? "quarantined" : "not quarantined"}"
+
+      quarantine_status
+    end
+
+    def status(file, command = SystemCommand)
+      command.run("/usr/bin/xattr",
+                    args:        ["-p", QUARANTINE_ATTRIBUTE, file],
+                    print_stderr: false).stdout
+    end
+
+    def cask(cask, downloaded_path, command = SystemCommand)
+      odebug "Quarantining #{downloaded_path}"
+      quarantiner = command.run(swift, args: ["#{HOMEBREW_LIBRARY_PATH}/cask/lib/hbc/utils/quarantine.swift", downloaded_path, cask.url.to_s, cask.homepage.to_s])
+
+      raise CaskError, "when quarantining #{downloaded_path}: #{quarantiner.stderr}" unless quarantiner.success?
```

Comment:
I may be overly pedantic here but to me, guard clauses that don’t start before column 80 are confusing and super easy to overlook.

Consider rewriting this in the if-return-else-raise form.